contents/tutorials/ruby-on-rails-analytics.md

```diff
@@ -0,0 +1,345 @@
+---
+title: How to set up Ruby on Rails analytics, feature flags and more
+date: 2023-02-02
+author: ["ian-vanagas"]
+showTitle: true
+sidebar: Docs
+featuredImage: ../images/tutorials/banners/ruby-on-rails-analytics.png
+topics: ["configuration", "feature flags", "persons", "events"]
+---
+
+Ruby on Rails is a popular fullstack web framework used by companies like Shopify, GitHub, Twitch, and more.
+
+In this tutorial, we build a basic Ruby on Rails app, add PostHog to it, and then set up some of the key features of PostHog like custom event capture, user identification, and feature flags.
+
+> Already know how to build a Ruby on Rails app? Skip to the [PostHog integration](#2-integrating-the-posthog-snippet-and-ruby-library).
+
+## 1. Creating a basic Ruby on Rails app
+
+First, install Ruby. I used `rbenv` and a `~/.bash_profile` to install version `3.1.3` but there are many ways you can do it. After installing the right version of Ruby, install Rails.
+
+```bash
+gem install rails
+```
+
+Once done, we can start our blog by running the `rails new` command. going into the blog folder, and starting the server.
+
+```bash
+rails new blog
+cd blog
+bin/rails server
+```
+
+Going to `localhost:3000` pulls up a site that looks like this:
+
+![App](../images/tutorials/ruby-on-rails-analytics/app.png)
+
+> A more detailed version of this Ruby app building process is in the [Getting Started with Rails](https://guides.rubyonrails.org/getting_started.html) guide.
+
+### Setting up the articles routes
+
+Rails is nice because it has a bunch of commands you can use to generate parts of the app. Use `generate controller` to create the controller for our articles:
+
+```bash
+bin/rails generate controller Articles index --skip-routes
+```
+
+We skip the routes because we set those up manually. In `config/routes.rb`, add the details to our route. The `resources` method lets us define a bunch of routes we want all a once:
```

Comment:
```suggestion
We skip the routes because we will set those up manually. In `config/routes.rb`, add details to the route. The `resources` method lets us define a bunch of routes we want all a once:
```